### PR TITLE
#1708 - Replace original implementation and remove copyright info from license file

### LINKED ISSUE #1708: Replace Drools CsvLineParser.java implementation
File: [drools-decisiontables/src/main/java/org/drools/decisiontable/parser/csv/CsvLineParser.java](https://github.com/apache/incubator-kie-drools/blob/fa989e50f0e71e2bd9d54adcb7d1781a41c27ace/drools-decisiontables/src/main/java/org/drools/decisiontable/parser/csv/CsvLineParser.java)

This would help remove the following reference in LICENSE file:

https://github.com/apache/incubator-kie-drools/blob/main/LICENSE#L238-L247

### PR DESCRIPTION
**Pull request by porcelli**
Closes: https://github.com/apache/incubator-kie-issues/issues/1708

### CODE REVIEW

**Review by tkobayas — APPROVED** [review]
Thank you!

**Comment by tkobayas** [comment]
kogito-apps GHA test failure can be ignored as it's being worked on.

File: LICENSE [hunk 1]
```diff
@@ -234,17 +234,6 @@ for drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/Ja
  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
  THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
- ------------------------------------------------------------------------------------
- for drools-decisiontables/src/main/java/org/drools/decisiontable/parser/csv/CsvLineParser.java
```
Comment (jomarko) [inline LICENSE:238]: Please, update also https://github.com/apache/incubator-kie-drools/blob/main/.rat-excludes, we need to remove the entry `CsvLineParsrer`  from there

File: drools-decisiontables/src/main/java/org/drools/decisiontable/parser/csv/CsvLineParser.java [hunk 2]
```diff
@@ -1,150 +1,93 @@
...
+    public List<String> parse(CharSequence input) {
+        String line = input != null ? input.toString() : "";
+        List<String> fields = new ArrayList<>();
 
-        /** Construct a CSV parser, with the default separator (','). */
-        public CsvParserImpl() {
-            this( CsvParserImpl.DEFAULT_SEP );
+        if (line.isEmpty()) {
+            fields.add("");
+            return fields;
```
Comment (jomarko) [inline drools-decisiontables/src/main/java/org/drools/decisiontable/parser/csv/CsvLineParser.java]: In past, we used often `Collections.singletoneList`, what is your opinion about:
```
        if (line.isEmpty()) {
            return Collections.singletonList("");
        }
```

File: drools-decisiontables/src/main/java/org/drools/decisiontable/parser/csv/CsvLineParser.java [hunk 3]
```diff
@@ -1,150 +1,93 @@
...
-     * This is insurance incase I need to replace it with more complex Csv
-     * handlers in the future.
+     * Creates a CSV parser with a custom delimiter.
+     *
+     * @param delimiter The character to use as field delimiter
      */
-    static interface ICsvParser {
-        public List<String> parse(String line);
+    public CsvLineParser(char delimiter) {
+        this.delimiter = delimiter;
```
Comment (jomarko) [inline drools-decisiontables/src/main/java/org/drools/decisiontable/parser/csv/CsvLineParser.java]: Do we want to keep this constructor? I was not able to find its usage, to me it seems we use only the default one.

**Review by jomarko — CHANGES_REQUESTED** [review]
Thank you for the PR @porcelli . Always good to see we are able to simplify the codebase we maintain.

I have some comments and questions before merge.

**Review by mariofusco — APPROVED** [review]

File: drools-decisiontables/src/main/java/org/drools/decisiontable/parser/csv/CsvLineParser.java (hunk 3 above)
Comment (porcelli) [reply drools-decisiontables/src/main/java/org/drools/decisiontable/parser/csv/CsvLineParser.java]: good point, I wasn't completely sure and that's why cheated and sent this way so to have the CI tell me that if in the codebase we had this used somehow :)

File: drools-decisiontables/src/main/java/org/drools/decisiontable/parser/csv/CsvLineParser.java (hunk 2 above)
Comment (porcelli) [reply drools-decisiontables/src/main/java/org/drools/decisiontable/parser/csv/CsvLineParser.java]: sure, I'll adjust

File: LICENSE (hunk 1 above)
Comment (porcelli) [reply LICENSE:238]: good catch

**Review by jomarko — APPROVED** [review]
Thank you for updates @porcelli

**Pull request merged**